daskdev-sat/app.py

```diff
@@ -50,10 +55,39 @@ def make_cluster(n_workers):
 class SaturnKubeCluster(KubeCluster):
     """Class that inherits from dask-kubernetes cluster"""
     def __init__(self, *args, dashboard_link=None, **kwargs):
-        """Init as usual, but add dashboard_link to object"""
+        """Init as usual, but add dashboard_link to object and start workers"""
         super().__init__(*args, **kwargs)
+
         self._dashboard_link = dashboard_link
 
+    async def _start(self):
+        if self._n_workers > 0:
+            name = self._generate_name
+            namespace = self._namespace
+            scheduler_address = f"tcp://{name}.{namespace}:{SCHEDULER_PORT}"
+            self.scheduler = Scheduler(
+                idle_timeout=self._idle_timeout,
+                service_wait_timeout_s=self._scheduler_service_wait_timeout,
+                core_api=kubernetes.client.CoreV1Api(),
+                pod_template=self.scheduler_pod_template,
+                namespace=namespace,
+            )
+            self.scheduler.address = scheduler_address
+            self.scheduler_comm = dask_rpc(
+                scheduler_address,
+                connection_args=self.security.get_connection_args("client"),
+            )
+            self.pod_template.spec.containers[0].env.append(
+                kubernetes.client.V1EnvVar(
+                    name="SATURN_DASK_SCHEDULER_ADDRESS", value=scheduler_address
+                )
```

Comment:
This the part I'm not entirely happy, but I fail to see another way.

dask-kubernetes core sets worker spec like that by [using `DASK_SCHEDULER_ADDRESS` env var ](https://github.com/dask/dask-kubernetes/blob/fdb77c0586fc83fabe3eb3059dd6f6797db0a2fd/dask_kubernetes/core.py#L132)with value set to `self.scheduler` which is passed by dask-distributed [`SpecCluster` package with `self.scheduler.address`](https://github.com/dask/distributed/blob/f761dd21e8cbba62273ab850f5f83e8a45a2fffa/distributed/deploy/spec.py#L348), however for some reason that is getting `None` once worker pods are started.

Reply:
hmmmm ok I think I understand, that makes sense